pkg/datapath/linux/node.go

```diff
@@ -73,7 +73,8 @@ type linuxNodeHandler struct {
 	// Node-scoped unique IDs for the nodes.
 	nodeIDsByIPs map[string]uint16
 
-	ipsecMetricCollector prometheus.Collector
+	ipsecMetricCollector  prometheus.Collector
+	ipsecMetricRegistered bool
```

Comment:
Might be simpler to just use a `sync.Once` and avoid changing the flow by adding cases for the new boolean

Reply:
Did you consider the dynamic of node configuration changing where ipsec is disabled/enabled? The straightforward solution with syn.Once would be:
- define a new helper that registers the metric only once, call that in the places where we register it now
- never unregister the metric

Reply:
Now that I reflect on this, I think I only hit the double registration issue in the integration tests due to other errors I had in the hive/cell changes. So maybe I can drop this commit from this PR? But it was confusing debugging those other issues when I saw the failures due to double registration of the ipsec metric!